test/index.js

```diff
@@ -1,223 +1,420 @@
-var nlb = require('../index.js'),
-    path = require('path'),
-    http = require('http'),
-    fs = require('fs'),
-    tap = require('tap');
+//moved this from package.json
+//some libraries behave better in NODE_ENV=test
+process.env.NODE_ENV='test';
+
+var request = require('supertest');
+var path = require('path');
+var recluster = require('../');
+//cluster is a singleton
+var cluster = require('cluster');
+var m = require('object-assign');
+var assert = require('assert');
+var fs = require('fs-extra');
+
+function passthru() {}
+
+var defaultOptions = {
+    respawn: 0.1,
+    workers: 2,
+    timeout: 1,
+    //don't spam the terminal :o
+    logger: {
+        log: passthru
+    }
+};
 
+// global handler to quickly spin up recluster
+var handler;
+
+// dry
+function requestExpectHelloWorld200(cb) {
+    request.get('/')
+        .expect(200)
+        .end(function(err, res) {
+            if (err) throw new Error(err);
+            assert.equal(res.text, 'hello world\n');
+            if (cb) cb();
+        });
+}
 
+function requestExpectError(cb) {
+    request.get('/')
+        .end(function(err, res) {
+            assert(err instanceof Error, 'err should be an Error');
+            assert.equal(err.message, 'connect ECONNREFUSED 127.0.0.1:8000');
+            if (cb) cb();
+        });
+}
 
-var lib = require('./lib/index'),
-    runTest = lib.runTest,
-    request = lib.request;
+/*
+We aren't supposed to do fancy things with this module, so calmly wait
+for everything to be cleanup up instead of rushing, it avoids all mistakes
 
+@fixme: this is probably a code smell, we should be able to have a listener when everything
+is cleaned up on recluster exposed object
+ */
+function cleanup(cb) {
+    if (Object.keys(cluster.workers).length > 0) {
+        cluster.on('exit', function exitHandler() {
+            if (Object.keys(cluster.workers).length !== 0) return;
+            cluster.removeListener('exit', exitHandler);
+            cb();
+        });
+        return handler.terminate();
+    }
+    cb();
```

Comment:
I don't think so, you are proxying the 'exit' calls from the workers, i'm simply waiting until the last one exit, and just return until there's none.

And the cb at the end was because for one test I actually called cleanup as a standard operating procedure when actually already was already cleanup.

Didn't want any tests to NOT call this cleanup, so I also handle the case when there are no workers and so I simply callback.